### PR TITLE
add getenv() function

### PR DESCRIPTION
**Pull request by drdzyk**
add [getenv()](https://www.php.net/manual/en/function.getenv.php) function

### CODE REVIEW

File: runtime/env.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+#include "runtime/env.h"
+
+static bool valid_environment_name(const char *name, const char *end) noexcept {
+  for (const char *s = name; s < end; ++s) {
+    if (*s == ' ' || *s == '.' || *s == '[') {
+      return false;
+    }
+  }
+  return true;
```
Comment (mkornaukhov03) [inline runtime/env.cpp:16]: Wouldn't 
```c++
return std::find_if(name, end, [](char s) {
  return vk::any_of_equal(s, ' ', '.', '[');
} == end;
```
be more compact and idiomatic?

Comment (drdzyk) [reply runtime/env.cpp]: true, it's done

File: builtin-functions/_functions.txt [hunk 2]
```diff
@@ -1614,3 +1614,5 @@ class DateTimeImmutable implements DateTimeInterface {
   public function getOffset(): int;
   public function getTimestamp(): int;
 }
+
+function getenv(string $varname = '', bool $local_only = false): mixed;
```
Comment (mkornaukhov03) [inline builtin-functions/_functions.txt:1618]: Why do you define return type as `mixed`, meanwhile the return type is `string | false` aka `Optional<String>`?
Why don't you add `getenv() : array` to this file?
Could explain these moments, please?

**Comment by mkornaukhov03** [comment]
Maybe we should add some C++ test for some corner-cases like empty `environment`, invalid `environment`, null `environment` etc, what do you think?

File: runtime/env.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+extern char **environ;
+#endif
+
+array<string> f$getenv() noexcept {
+  array<string> res;
+  for (char **env = environ; env && *env; ++env) {
+    const char *ptr = std::strchr(*env, '=');
+    // malformed entry?
+    if (ptr && ptr != *env && valid_environment_name(*env, ptr)) {
+      const string key{*env, static_cast<uint32_t>(ptr - *env)};
```
Comment (mkornaukhov03) [inline runtime/env.cpp:24]: Maybe `std::distance(*env, ptr) ` would be more idiomatic

File: builtin-functions/_functions.txt (hunk 2 above)
Comment (drdzyk) [reply builtin-functions/_functions.txt:1618]: Because I can't overload function in php.

**Comment by drdzyk** [comment]
> Maybe we should add some C++ test for some corner-cases like empty `environment`, invalid `environment`, null `environment` etc, what do you think?

I guess that these functions are quite simple and phpt tests will suffice.

File: runtime/env.cpp (hunk 3 above)
Comment (drdzyk) [reply runtime/env.cpp:24]: Let me leave this little thing as is please

**Review by mkornaukhov03 — APPROVED** [review]
LGTM
